zipkin-collector/core/src/main/java/zipkin2/collector/Collector.java

```diff
@@ -195,19 +213,34 @@ String idString(Span span) {
     return sampled;
   }
 
-  Callback<Void> storeSpansCallback(final List<Span> spans) {
-    return new Callback<Void>() {
-      @Override public void onSuccess(Void value) {
-      }
+  class StoreSpans implements Callback<Void>, Runnable {
+    final List<Span> spans;
 
-      @Override public void onError(Throwable t) {
-        handleStorageError(spans, t, NOOP_CALLBACK);
-      }
+    StoreSpans(List<Span> spans) {
+      this.spans = spans;
+    }
 
-      @Override public String toString() {
-        return appendSpanIds(spans, new StringBuilder("StoreSpans(")) + ")";
+    @Override public void run() {
+      try {
+        store(spans, this);
+      } catch (RuntimeException | Error e) {
+        // While unexpected, invoking the storage command could raise an error synchronously. When
+        // that's the case, we wouldn't have invoked callback.onSuccess, so we need to handle the
+        // error here.
+        onError(e);
       }
-    };
+    }
+
+    @Override public void onSuccess(Void value) {
+    }
+
+    @Override public void onError(Throwable t) {
+      handleStorageError(spans, t, NOOP_CALLBACK);
```

Comment:
the intent here is to only log on error, and that's why we use noop-callback as opposed to passing in the callback from the transport. Otherwise, we would hold up the request.